### PR TITLE
Implement error scrubbing middelware with opts

### PR DESCRIPTION
**Pull request by LucasRoesler**
**What**
- Add new Option to scrub grpc error messages
- Allow the user to configure the scrubbing behavior using an optional
  function parameter
- Update the unit tests
- Add an "Err" method on the test proto server

**Notes**
- Had issues with vendoring and go 1.11 , which is why the aes package is copied in the way it is

### CODE REVIEW

File: grpc/errors.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// WithErrorScrubbing ensures that internal or unknown errors do not leak information.
+// You  may pass a function to customize the scrubbing behavior. The package provides
+// two simple scrubbers: SimpleErrorScrubber and NoopErrorScrubber for convenience.
```
Comment (trusch) [inline grpc/errors.go]: include encryting scrubber in comment

File: grpc/errors.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		st := status.New(codes.Internal, "Internal Server Error")
+		return st.Err()
+	default:
+		return err
+	}
+}
+
+// NewEncryptedErrorScrubber replaces unknown or internal errors with a generic error object. The
+// original error message is encrypted and added to the error message in the status.Details
+func NewEncryptedErrorScrubber(key string) func(error) error {
```
Comment (trusch) [inline grpc/errors.go]: remove the `New` prefix to be uniform with the other two scrubbers

File: grpc/errors_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
+package grpc
```
Comment (trusch) [inline grpc/errors_test.go:1]: add a test for SimpleErrorScrubber and NewEncryptedErrorScrubber

**Review by trusch — CHANGES_REQUESTED** [review]

File: grpc/errors_test.go (hunk 3 above)
Comment (LucasRoesler) [reply grpc/errors_test.go:1]: SimpleErrorScrubber is the default test

I will add one for the encrypted

File: grpc/errors.go (hunk 2 above)
Comment (LucasRoesler) [reply grpc/errors.go]: But it is a scrubber factory

File: grpc/errors_test.go (hunk 3 above)
Comment (LucasRoesler) [reply grpc/errors_test.go:1]: done

**Comment by LucasRoesler** [comment]
@trusch i addressed each comment

**Review by trusch — APPROVED** [review]

**Pull request merged**